### PR TITLE
Expenses - Fix Deletion

### PR DESCRIPTION
**Pull request by cuddyz**
## What / Why

This change is built on top of the `yemi` branch 🚨 

This change updates the `expenseRoute.js` to reference the correct `Date` key and use the correct deletion method from mongoose.

### CODE REVIEW

File: routes/expenseRoute.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ router.post('/', (req, res)=>{
         type: req.body.type,
         amount: req.body.amount,
         description: req.body.description,
-        date: req.body.date
+        Date: req.body.Date
```
Comment (cuddyz) [inline routes/expenseRoute.js:11]: **note:** Similar to the Income issue earlier, Date needs to be capitalized

File: routes/expenseRoute.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ router.patch('/:id', getExpense, async (req, res) => {
 //Deleting One
 router.delete('/:id', getExpense, async (req, res) => {
     try {
-        await res.expense.remove()
+        await res.expense.findByIdAndRemove(req.params.id)
```
Comment (cuddyz) [inline routes/expenseRoute.js:59]: **note:** We need to be sure we are using the correct method from Mongoose.  `.remove()` drops the entire table 😱

File: routes/expenseRoute.js (hunk 1 above)
Comment (omdnrd) [reply routes/expenseRoute.js:11]: oh wtf, i thought I got that one. thank you

File: routes/expenseRoute.js (hunk 2 above)
Comment (omdnrd) [reply routes/expenseRoute.js:59]: ahhh okay that makes sense. I'm only just realizing how many different commands mongoose comes with

Comment (omdnrd) [reply routes/expenseRoute.js:59]: So I'm thinking for patch right, would I just do it in a similar way as the delete button with find by ID but just call the patch function instead? But also I'd use some sort of display box so the user can see what they're editing? Or what would be the best way to go about editing?

Comment (cuddyz) [reply routes/expenseRoute.js:59]: Yea it sounds like you are on the right track @omdnrd.  

What I'd suggest:

1. Add someplace in the UI for a user to "edit" an existing one
2. Let them click "save" once they finish their edit and that will call the update method in the `.ts` controller. 
3. Ensure the `.ts` controller calls the proper service file's method.
4. Ensure that the `service file` method does what the delete method does by sending down the `id` and any data in the body that needs to be updated.
5. Ensure that the server patch method does what you expect it to.

You should go ahead and give this a shot before our call at 11 and see how far you can get.

**Pull request merged**